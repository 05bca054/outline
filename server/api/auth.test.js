import TestServer from 'fetch-test-server';
import app from '..';
import { flushdb, sequelize, seed } from '../test/support';

const server = new TestServer(app.callback());

beforeEach(flushdb);
afterAll(() => server.close());
afterAll(() => sequelize.close());

it('should signup a new user', async () => {
  const res = await server.post('/api/auth.signup', {
    body: {
      username: 'testuser',
      name: 'Test User',
      email: 'new.user@example.com',
      password: 'test123!',
    },
  });
  const body = await res.json();

  expect(res.status).toEqual(200);
  expect(body.ok).toBe(true);
  expect(body.data.user).toBeTruthy();
});

it('should require params', async () => {
  const res = await server.post('/api/auth.signup', {
    body: {
      username: 'testuser',
    },
  });
  const body = await res.json();

  expect(res.status).toEqual(400);
  expect(body).toMatchSnapshot();
});


it('should require valid email', async () => {
  const res = await server.post('/api/auth.signup', {
    body: {
      username: 'testuser',
      name: 'Test User',
      email: 'example.com',
      password: 'test123!',
    },
  });
  const body = await res.json();

  expect(res.status).toEqual(400);
  expect(body).toMatchSnapshot();
});

it('should require unique email', async () => {
  await seed();
  const res = await server.post('/api/auth.signup', {
    body: {
      username: 'testuser',
      name: 'Test User',
      email: 'user1@example.com',
      password: 'test123!',
    },
  });
  const body = await res.json();

  expect(res.status).toEqual(400);
  expect(body).toMatchSnapshot();
});

it('should require unique username', async () => {
  await seed();
  const res = await server.post('/api/auth.signup', {
    body: {
      username: 'user1',
      name: 'Test User',
      email: 'userone@example.com',
      password: 'test123!',
    },
  });
  const body = await res.json();

  expect(res.status).toEqual(400);
  expect(body).toMatchSnapshot();
});
