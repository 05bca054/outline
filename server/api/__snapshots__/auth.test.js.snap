exports[`#auth.signup should require params 1`] = `
Object {
  "error": "name is required",
  "ok": false
}
`;

exports[`#auth.signup should require unique email 1`] = `
Object {
  "error": "User already exists with this email",
  "ok": false
}
`;

exports[`#auth.signup should require unique username 1`] = `
Object {
  "error": "User already exists with this username",
  "ok": false
}
`;

exports[`#auth.signup should require valid email 1`] = `
Object {
  "error": "email is invalid",
  "ok": false
}
`;

exports[`#login should login with email 1`] = `
Object {
  "avatarUrl": "http://example.com/avatar.png",
  "id": "86fde1d4-0050-428f-9f0b-0bf77f8bdf61",
  "name": "User 1",
  "username": "user1"
}
`;

exports[`#login should login with username 1`] = `
Object {
  "avatarUrl": "http://example.com/avatar.png",
  "id": "86fde1d4-0050-428f-9f0b-0bf77f8bdf61",
  "name": "User 1",
  "username": "user1"
}
`;

exports[`#login should require either username or email 1`] = `
Object {
  "error": "username or email is required",
  "ok": false
}
`;

exports[`#login should require password 1`] = `
Object {
  "error": "password is required",
  "ok": false
}
`;

exports[`#login should validate password 1`] = `
Object {
  "error": "Invalid password",
  "ok": false
}
`;
